core/src/main/scala/kafka/server/metadata/BrokerMetadataListener.scala

```diff
@@ -244,6 +248,9 @@ class BrokerMetadataListener(
       _publisher = Some(publisher)
       log.info(s"Starting to publish metadata events at offset $highestMetadataOffset.")
       try {
+        if (shouldSnapshot()) {
```

Comment:
1. Yes, a listener(here we refer to BrokerMetadataListener) will potentially receive a handleCommit or a handleSnapshot every time when it receives a metadata log change(for example, create a topic), when a broker startup, it will not start publishing until it gets all the metadata logs(I call this period "Catching up period" because a broker needs to connect to the controller and fetch all metadata logs).
2. We used to create a snapshot if it detects a metadata version change or _bytesSinceLastSnapshot is bigger enough.
3. But I changed the logic here to not create a snapshot in "Catching up period", so when starting publishing, which means the end of "Catching up period", we should inspect that is it necessary to create a snapshot.